packages/frontend/src/features/sqlRunner/components/CartesianChartStyling.tsx

```diff
@@ -19,6 +20,24 @@ export const CartesianChartStyling = ({
 
     const currentConfig = useAppSelector(selectCurrentCartesianChartState);
 
+    const series = useAppSelector((state) => {
+        if (
+            !state.barChartConfig.config?.fieldConfig?.y ||
+            state.barChartConfig.config.fieldConfig.y.length <= 1
+        ) {
+            return [];
+        }
+        return state.barChartConfig.config.fieldConfig.y.map((f) => {
+            const format =
+                state.barChartConfig.config?.display?.series?.[f.reference]
+                    ?.format;
+            return {
+                reference: f.reference,
+                format,
+            };
+        });
```

Comment:
should I create its own selector?

Reply:
Probably? The advantage of selectors is that they can be very specific and you can have them only update the selected value when that value changes. If you do it this way, I think this component will re-render more than necessary